django/contrib/auth/forms.py

```diff
@@ -178,16 +178,26 @@ def __init__(self, request=None, *args, **kwargs):
         self.user_cache = None
         super(AuthenticationForm, self).__init__(*args, **kwargs)
 
-        # Set the label for the "username" field.
+        # Set the label and maxlength for the "username" field.
         UserModel = get_user_model()
         self.username_field = UserModel._meta.get_field(UserModel.USERNAME_FIELD)
+        if self.username_field.max_length is not None:
+            self.fields['username'].max_length = self.username_field.max_length
+            self.fields['username'].widget.attrs['maxlength'] = self.username_field.max_length
         if self.fields['username'].label is None:
             self.fields['username'].label = capfirst(self.username_field.verbose_name)
 
+    def clean_username(self):
+        username = self.cleaned_data.get('username')
+        if len(username) > self.username_field.max_length:
```

Comment:
This is normal condition on clean method,for check greater length username  of 150 characters,